src/definitions.jl

```diff
@@ -396,6 +396,49 @@ function ifftshift(x,dim)
     circshift(x, s)
 end
 
+##############################################################################
+
+
+struct Frequencies{T<:Number} <: AbstractVector{T}
+    nreal::Int
+    n::Int
+    multiplier::T
+end
+
+unsafe_getindex(x::Frequencies, i::Int) =
+    (i-1+ifelse(i <= x.nreal, 0, -x.n))*x.multiplier
+function Base.getindex(x::Frequencies, i::Int)
+    (i >= 1 && i <= x.n) || throw(BoundsError())
+    unsafe_getindex(x, i)
+end
+
+function Base.iterate(x::Frequencies, i::Int=1)
+    i > x.n ? nothing : (unsafe_getindex(x,i), i + 1)
+end
+Base.size(x::Frequencies) = (x.n,)
+Base.step(x::Frequencies) = x.multiplier
+
+"""
+    fftfreq(n, fs=1)
+Return discrete fourier transform sample frequencies. The returned
+Frequencies object is an AbstractVector containing the frequency
+bin centers at every sample point. `fs` is the sample rate of the
+input signal.
+"""
+fftfreq(n::Int, fs::Number=1) = Frequencies(((n-1) >> 1)+1, n, fs/n)
```

Comment:
`(n+1) >> 1` is equivalent and seems simpler.

Reply:
If we want the first argument to be the number of nonnegative frequencies, then I think it should be `(n+3) >> 1`